assets/js/googlesitekit/datastore/user/surveys.js

```diff
@@ -113,28 +190,41 @@ const baseActions = {
 				return {};
 			}
 
-			const cacheKey = createCacheKey( 'core', 'user', 'survey-trigger', {
+			// Await for surveys to be resolved before checking timeouts.
+			yield Data.commonActions.await(
+				__experimentalResolveSelect( CORE_USER ).getSurveyTimeouts()
+			);
+
+			const isTimedOut = select( CORE_USER ).isSurveyTimedOut(
+				triggerID
+			);
+			const isTimingOut = select( CORE_USER ).isTimingOutSurvey(
 				triggerID,
-			} );
-			const { cacheHit } = yield Data.commonActions.await(
-				getItem( cacheKey )
+				ttl
 			);
 
-			if ( false === cacheHit ) {
+			if ( ! isTimedOut && ! isTimingOut ) {
```

Comment:
Should these be checks for `isTimedOut === false` rather than `! isTimedOut`? Because `isTimedOut` could be `undefined`… do we want this to behave the same way if it's `false` and `undefined`?

```suggestion
			if ( ! isTimedOut && ! isTimingOut ) {
```

Reply:
Those variables are always resolved by the time we get them. Added a comment that explains it.

Reply:
Ah, awesome, thanks! I think in that case too it's fine to be less explicit with the checks as anyone reading this for prior art will see the comment and know why we're doing a false-y rather than `=== false` check. 👍🏻